compiler/daml-lf-verify/tests/DA/Daml/LF/Verify/Tests.hs

```diff
@@ -0,0 +1,97 @@
+  [ testCase "Iou_Split" $ do
+      quickstartDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/quickstart.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "Iou_Split"
+          field = FieldName "amount"
+      result <- verify quickstartDar False tmpl choice tmpl field
+      assertEqual "Verification failed for Iou_Split - amount"
+        Success result
+  , testCase "Iou_Merge" $ do
+      quickstartDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/quickstart.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "Iou_Merge"
+          field = FieldName "amount"
+      result <- verify quickstartDar False tmpl choice tmpl field
+      assertEqual "Verification failed for Iou_Merge - amount"
+        Success result
+  ]
+
+conditionalTests :: TestTree
+conditionalTests = testGroup "Conditionals"
+  [ testCase "Success A" $ do
+      condDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/conditionals.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "SuccA"
+          field = FieldName "content"
+      result <- verify condDar False tmpl choice tmpl field
+      assertEqual "Verification failed for SuccA - content"
+        Success result
+  , testCase "Success B" $ do
+      condDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/conditionals.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "SuccB"
+          field = FieldName "content"
+      result <- verify condDar False tmpl choice tmpl field
+      assertEqual "Verification failed for SuccB - content"
+        Success result
+  , testCase "Success C" $ do
+      condDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/conditionals.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "SuccC"
+          field = FieldName "content"
+      result <- verify condDar False tmpl choice tmpl field
+      assertEqual "Verification failed for SuccC - content"
+        Success result
+  , testCase "Success D" $ do
+      condDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/conditionals.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "SuccD"
+          field = FieldName "content"
+      result <- verify condDar False tmpl choice tmpl field
+      assertEqual "Verification failed for SuccD - content"
+        Success result
+  , testCase "Fail A" $ do
+      condDar <- locateRunfiles (mainWorkspace </> "compiler/daml-lf-verify/data/conditionals.dar")
+      let tmpl = TypeConName ["Iou"]
+          choice = ChoiceName "FailA"
+          field = FieldName "content"
+      verify condDar False tmpl choice tmpl field >>= \case
+        Success -> assertFailure "Verification wrongfully passed for FailA - content"
+        Unknown -> assertFailure "Verification inconclusive for FailA - content"
+        Fail _ -> return ()
```

Comment:
Ar the results not deterministic or why are you not just using `assertEqual` on the full `Fail` expression?